R/randomForest_tidiers.R

```diff
@@ -811,3 +834,142 @@ calc_feature_imp <- function(df,
 
   do_on_each_group(clean_df, each_func, name = "model", with_unnest = FALSE)
 }
+
+#' @export
+#' @param type "importance", "evaluation" or "conf_mat". Feature importance, evaluated scores or confusion matrix of training data.
+tidy.ranger <- function(x, type = "importance", pretty.name = FALSE, ...) {
+  switch(
+    type,
+    importance = {
+      # return variable importance
+
+      imp <- ranger::importance(x)
+
+      ret <- data.frame(
+        variable = x$terms_mapping[names(imp)],
+        importance = imp,
+        stringsAsFactors = FALSE
+      )
+
+      ret
+    },
+    evaluation = {
+      # get evaluation scores from training data
+      actual <- x$y
+
+      if(is.numeric(actual)){
+        glance(x, pretty.name = pretty.name, ...)
+      } else {
+        predicted <- x$predictions
+
+        per_level <- function(class) {
+          tp <- sum(actual == class & predicted == class, na.rm = TRUE)
+          tn <- sum(actual != class & predicted != class, na.rm = TRUE)
+          fp <- sum(actual != class & predicted == class, na.rm = TRUE)
+          fn <- sum(actual == class & predicted != class, na.rm = TRUE)
+
+          precision <- tp / (tp + fp)
+          # this avoids NA
+          if(tp+fp == 0) {
+            precision <- 0
+          }
+
+          recall <- tp / (tp + fn)
+          # this avoids NA
+          if(tn+fn == 0) {
+            recall <- 0
+          }
+
+          accuracy <- (tp + tn) / (tp + fp + tn + fn)
+
+          f_score <- 2 * ((precision * recall) / (precision + recall))
+          # this avoids NA
+          if(precision + recall == 0) {
+            f_score <- 0
+          }
+
+          data_size <- sum(actual == class)
+
+          ret <- data.frame(
+            class,
+            f_score,
+            accuracy,
+            1- accuracy,
+            precision,
+            recall,
+            data_size
+          )
+
+          names(ret) <- if(pretty.name){
+            c("Class", "F Score", "Accuracy Rate", "Missclassification Rate", "Precision", "Recall", "Data Size")
+          } else {
+            c("class", "f_score", "accuracy_rate", "missclassification_rate", "precision", "recall", "data_size")
+          }
+          ret
+        }
+
+        if(x$classification_type == "binary") {
+          ret <- per_level(levels(actual)[2])
+          # remove class column
+          ret <- ret[, 2:6]
+          ret
+        } else {
+          dplyr::bind_rows(lapply(levels(actual), per_level))
+        }
+      }
+    },
+    conf_mat = {
+      # return confusion matrix
+      ret <- data.frame(
+        actual_value = x$y,
+        predicted_value = x$predictions
+      ) %>%
+        dplyr::filter(!is.na(predicted_value))
+
+      if (is.numeric(ret$actual_value)){
+        tile <- cut(c(ret$actual_value, ret$predicted_value), 10)
+        ret$actual_bucket <- tile[seq(length(tile)/2)]
+        ret$predicted_bucket <- tile[length(tile)/2 + seq(length(tile)/2)]
```

Comment:
This conf_mat for regression part seems to be unfinished.
Please either make it work or remove.